proxy/proxy.go

```diff
@@ -0,0 +1,53 @@
+package proxy
+
+import (
+	"log"
+	"net/http"
+	"sync"
+
+	"github.com/gorilla/websocket"
+	"github.com/taskcluster/webhooktunnel/wsmux"
+)
+
+type proxy struct {
+	m        sync.RWMutex
+	pool     map[string]*wsmux.Session
+	upgrader websocket.Upgrader
+}
+
+func NewProxy(upgrader websocket.Upgrader) *proxy {
```

Comment:
Why pass in the upgrader... just create it in the constructor... is there any reason consumers want to configure it?